torntools/scripts/content/stockexchange/ttStockexchange.js

```diff
@@ -232,191 +48,121 @@ function addFilter(filters) {
+		filterContainer.find(`#total-profit-filter input[type='checkbox'][value='${state}']`).checked = true;
 	}
-	if (isPortfolio) {
-		for (let state of filters.stock_exchange[filterPage].profitLoss) {
-			doc.find(`#extra-filter #profit-filter input[type='checkbox'][value='${state}']`).checked = true;
-		}
+	if (filters.stock_exchange.owned) filterContainer.find("#owned-filter input").checked = true;
+	if (filters.stock_exchange.benefit) filterContainer.find("#benefit-filter input").checked = true;
 
-		doc.find(`#extra-filter #listed-filter input`).value = filters.stock_exchange[filterPage].listedOnly;
-	}
-	doc.find(`#extra-filter #name-filter input`).value = filters.stock_exchange[filterPage].name;
+	filterContainer.find(`#extra-filter #name-filter input`).value = filters.stock_exchange.name;
 
 	applyFilters();
 
 	function applyFilters() {
-		let worth = [];
-		let forecast = [];
-		let profitLoss = [];
-		let listedOnly;
-
-		// Worth
-		for (let checkbox of doc.findAll("#worth-filter input[type='checkbox']:checked")) {
-			worth.push(checkbox.getAttribute("value"));
-		}
-		// Forecast
-		for (let checkbox of doc.findAll("#forecast-filter input[type='checkbox']:checked")) {
-			forecast.push(checkbox.getAttribute("value"));
-		}
+		let change = [];
+		let totalProfitLoss = [];
+		let owned, benefit;
+
 		const name = doc.find("#extra-filter #name-filter input").value;
-		if (isPortfolio) {
-			for (let checkbox of doc.findAll("#extra-filter #profit-filter input[type='checkbox']:checked")) {
-				profitLoss.push(checkbox.getAttribute("value"));
-			}
-			listedOnly = doc.find("#extra-filter #listed-filter input").checked;
+		for (let checkbox of doc.findAll("#change-filter input[type='checkbox']:checked")) {
+			change.push(checkbox.getAttribute("value"));
 		}
+		for (let checkbox of doc.findAll("#total-profit-filter input[type='checkbox']:checked")) {
+			totalProfitLoss.push(checkbox.getAttribute("value"));
+		}
+		owned = doc.find("#tt-stock-filter #owned-filter input[type='checkbox']").checked;
+		benefit = doc.find("#tt-stock-filter #benefit-filter input[type='checkbox']").checked;
 
 		// Filtering
-		for (let stock of doc.findAll(".stock-list > .item, .portfolio-list-shares > li.item-wrap")) {
-			const stockId = isPortfolio
-				? stock
-						.find(".logo > a")
-						.getAttribute("href")
-						.match(/&ID=([0-9]*)/i)[1]
-				: stock.firstElementChild.getAttribute("action").split("ID=")[1];
-
-			const data = torndata.stocks[stockId];
+		for (let stock of doc.findAll("#stockmarketroot [class*='stockMarket__'] > *:not(#panel-InfoTab, #panel-ManagerTab)")) {
+			const stockAcronym = stock.find("[class*='logoContainer__'] img").src.split("/").pop().replace(".svg", "");
+			const stockID = torndata.stocks.findIndex((stockEntry) => {
+				return stockEntry.acronym === stockAcronym;
+			});
+			const data = torndata.stocks[stockID];
+			const userStockData = userdata.stocks.filter((x) => {
```

Comment:
They changed the stocks api, so this is no longer correct.

Reply:
Done.